src/metrics/metrics.go

```diff
@@ -0,0 +1,45 @@
+package metrics
+
+import (
+	"context"
+	stats "github.com/lyft/gostats"
+	"google.golang.org/grpc"
+	"time"
+)
+
+type serverMetrics struct {
+	totalRequests stats.Counter
+	responseTime  stats.Timer
+}
+
+// ServerReporter reports server-side metrics for ratelimit gRPC server
+type ServerReporter struct {
+	scope stats.Scope
+}
+
+func newServerMetrics(scope stats.Scope, fullMethod string) *serverMetrics {
+	_, methodName := splitMethodName(fullMethod)
+	ret := serverMetrics{}
+	ret.totalRequests = scope.NewCounter(methodName + ".total_requests")
```

Comment:
I know the existing code tends to use metric names rather than tags, but for what its worth: I'd much rather see `methodName` as a tag rather than part of the metric name. (More generally, it seems like there's little to no reason a metric name ever needs to be dynamically created in a system that supports metrics with tags)

Using tags makes it simple in most metric systems to aggregate stats across method names or break down to specific subsets of method names.

Aggregating across metric names, by contrast, is not particularly simple or normal or standard in most metrics systems. Using different metric names also makes discovering what metric exist by looking at metric names a mess.

Reply:
I agree that metrics tags are always the better choice than embedding in metrics name, but I don't think prom-statsd exporter supports the tagging format used by gostats. (please correct me if I'm wrong) So we'll have to do regexp matching to export it as a prom label, which is slower and more work than using glob patterns.

If there is a quick and easy way to export tags as prom labels, I'm happy to use tags instead.